### PR TITLE
ci: added release workflow

### PR DESCRIPTION
**Pull request by d-roak**
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/933)

### CODE REVIEW

File: .github/workflows/deployments.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
       matrix:
         network: [testnet]
     env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (d-roak) [inline .github/workflows/deployments.yml:19]: u need to confirm if this will work. I don't know if the token permissions are shared across repos

Comment (ClementWalter) [reply .github/workflows/deployments.yml:19]: this is fed by the github runner

Comment (d-roak) [reply .github/workflows/deployments.yml:19]: yep, but you are trying to use the token to get artifacts from `kakarot-ssj` right?

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**